packages/wonder-blocks-button/components/button.js

```diff
@@ -136,6 +134,23 @@ export type SharedProps = {|
      * href is not
      */
     onClick?: (e: SyntheticEvent<>) => mixed,
+
+    /**
+     * Run async code before navigating. If the promise returned rejects then
+     * navigation will not occur.
+     *
+     * If both safeWithNav and beforeNav are provided, beforeNav will be run
+     * first and safeWithNav will only be run if beforeNav doesn't not reject.
+     */
+    beforeNav?: (e: SyntheticEvent<>) => Promise<mixed>,
+
+    /**
+     * Run async code in the background while client-side navigating. If the
+     * navigation is server-side, the callback must either be resolved or
```

Comment:
You could say: the promise must be settled...

(Which is a more accurate way to say what's happening)